src/connection.rs

```diff
@@ -0,0 +1,188 @@
+use byteorder::{BigEndian, ReadBytesExt};
+use openssl::ssl::*;
+use protobuf::CodedOutputStream;
+use std::io::{Cursor, Read};
+use std::net::TcpStream;
+use std::path::Path;
+use std::sync::Arc;
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+use message_types::MessageType::*;
+use message_types::*;
+use mumble;
+
+// Connection timeout in seconds. It waits this long if no message was received
+const TIMEOUT: u64 = 5;
+
+// How long to wait between pings
+const PING_DELAY: u64 = 10;
+
+/// A way to interact with the Mumble server
+pub struct Connection {
+    stream: SslStream<TcpStream>,
+    session: Option<u32>,
```

Comment:
Since most real functionality only works once a session is established: Could we split this into two types, one for a connection that is in the process of being initialized (which should probably not even be exposed to the user using the default API), and one for a connection that had it's session initialized and is ready to send text messages, change channel, etc?

Edit: Thinking about this some more, the connection should probably loose the session, user list, etc. and just be a way of sending and receiving messages. A separate `Session` struct that wraps a connection should hold session id, user lists, etc.

Reply:
This seems like a good idea. I'll make a new session struct.

Reply:
Ok, I just committed the change. Can you take a look?